### PR TITLE
Feature/137 select component

### PR DESCRIPTION
**Pull request by Diaan**

### CODE REVIEW

File: packages/core/src/avatar/avatar.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    super.connectedCallback();
+
+    this.user = await this._getUserDetails(this.uniqueProfileId);
+  }
+
+  async _getUserDetails(id: string): Promise<UserProfile | undefined> {
+    try {
+      const response = await fetch(`https://randomuser.me/api/?inc=picture,name&seed=slds-${id}`);
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-explicit-any
+      const json: { results: UserProfile[]; info: any } = await response?.json();
```
Comment (anna-lach) [inline packages/core/src/avatar/avatar.ts:52]: Maybe we can use `unknown` here instead of `any`?

File: packages/core/src/select/select-option.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  }
+
+  override connectedCallback(): void {
+    super.connectedCallback();
+    this.setAttribute('role', 'option');
+  }
+
+  override firstUpdated(): void {
+    this.#observer = new ResizeObserver(m => this.#handleResize(m));
+    this.#observer?.observe(this);
```
Comment (anna-lach) [inline packages/core/src/select/select-option.ts:50]: Maybe we should add `this.#observer?.disconnect()` on `disconnectedCallback()`?

File: packages/core/src/select/select-overlay.scss [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  --_max-overlay-height: var(--max-overlay-height, 80vh);
+}
+
+:host {
+  background-color: #fff;
+  display: flex;
+  flex-direction: column;
+  gap: 4px;
+  max-height: var(--_max-overlay-height);
+  overflow: auto;
```
Comment (anna-lach) [inline packages/core/src/select/select-overlay.scss:11]: Maybe we should remove it because we have `overflow-x` and `overflow-y` below?

**Comment by anna-lach** [comment]
It looks like exports for select and avatar components is missing in `package.json`?

File: packages/core/src/avatar/avatar.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
+import type { CSSResultGroup, TemplateResult } from 'lit';
+import { LitElement, html } from 'lit';
+import { property, state } from 'lit/decorators.js';
+import styles from './avatar.scss.js';
+
+export interface UserProfile {
+  name: {
+    first: string;
```
[the remark this answers is not in the document]
Comment (Diaan) [reply packages/core/src/avatar/avatar.ts:8]: This is actually a (very) temporary component, just to have something to render in the select, so it's not really very important, but I've changed it anyway

**Comment by anna-lach** [comment]
> It looks like exports for select and avatar components are missing in `package.json`?

Perhaps we also need to add exports for select option and select option group?

**Pull request merged**